src/auth/auth.service.ts

```diff
@@ -0,0 +1,16 @@
+import { Injectable } from '@nestjs/common';
+import { UsersService } from '../users/users.service';
+import { User } from '../users/user.entity';
+
+@Injectable()
+export class AuthService {
+  constructor(private usersService: UsersService) {}
+
+  async validateToken(token: string): Promise<User> {
+    const user = await this.usersService.getUserByAuthToken(token);
```

Comment:
As I stated in #738 the token should be hashed and this has effect on this line because it has to be hashed here.

Reply:
Which hashing algorithm should be used here? bcrypt?

Reply:
fixed with bcrypt.

Reply:
yes bcrypt should be fine I think.

Reply:
We will use sha256 because we don't need a salt and gives no benefit and makes it harder to get which user owns the auth token